modules/data.atmosphere/R/met2cf.ERA5.R

```diff
@@ -0,0 +1,171 @@
+      #For ERA5 daily ensemble data, the accumulation period is 3 hours. Hence to convert to W/m2:
+      ens[, "ssrd"] <- ens[, "ssrd"]/ (3 * 3600)
+      ens[, "strd"] <- ens[, "strd"]/ (3 * 3600)
+      #precipitation it's originaly in meters. Meters times the density will give us the kg/m2
+      ens[, "tp"] <- ens[, "tp"]*1000/3 # divided by 3 because we have 3 hours data
+      ens[, "tp"] <- udunits2::ud.convert(ens[, "tp"], "kg m-2 hr-1", "kg m-2 6 s-1")  #There are 21600 seconds in 6 hours
+      #RH
+      #Adopted from weathermetrics/R/moisture_conversions.R
+      t <- udunits2::ud.convert(ens[, "t2m"] %>% as.numeric(), "K", "degC")
+      dp <- udunits2::ud.convert(ens[, "d2m"] %>% as.numeric(), "K", "degC")
+      beta <- (112 - (0.1 * t) + dp) / (112 + (0.9 * t))
+      relative.humidity <- 100 * beta ^ 8
+      #specific humidity
+      specific_humidity <- PEcAn.data.atmosphere::rh2qair(relative.humidity,
+                                                          ens[, "t2m"] %>% as.numeric(),
+                                                          ens[, "sp"] %>% as.numeric()
+                                                          ) # Pressure in Pa
+
+      #adding humidity
+      xts::merge.xts(ens[,-c(3)], (specific_humidity)) %>%
+        `colnames<-`(
+          c(
+            "air_temperature",
+            "air_pressure",
+            "precipitation_flux",
+            "eastward_wind",
+            "northward_wind",
+            "surface_downwelling_shortwave_flux_in_air",
+            "surface_downwelling_longwave_flux_in_air",
+            "specific_humidity"
+          )
+        )
+      
+    })
+  
+  
+  #These are the cf standard names
+  cf_var_names = colnames(out.new[[1]])
+  cf_var_units = c("K", "Pa", "kgm-2s-1", "ms-1", "ms-1", "Wm-2", "Wm-2", "1")  #Negative numbers indicate negative exponents
+  
+  # Create a data frame with information about the file.  This data frame's format is an internal PEcAn standard, and is stored in the BETY database to
+  # locate the data file. 
+  results <- data.frame(
+    file = "", #Path to the file (added in loop below).
+    host = PEcAn.remote::fqdn(), 
+    mimetype = "application/x-netcdf", 
+    formatname = "CF Meteorology", 
+    startdate = paste0(format(start_date, "%Y-%m-%dT%H:%M:00")), 
+    enddate = paste0(format(end_date, "%Y-%m-%dT%H:%M:00")), 
+    dbfile.name = "ERA5", 
+    stringsAsFactors = FALSE
+  )
+  
+  #Each ensemble gets its own file.
+  time_dim = ncdf4::ncdim_def(
+    name = "time",
+    paste(units = "hours since", format(start_date, "%Y-%m-%dT%H:%M")),
+    seq(0, (length(index(out.new[[1]])) * 3) - 1 , length.out = length(index(out.new[[1]]))),
+    create_dimvar = TRUE
+  )
+  lat_dim = ncdf4::ncdim_def("latitude", "degree_north", lat, create_dimvar = TRUE)
+  lon_dim = ncdf4::ncdim_def("longitude", "degree_east", long, create_dimvar = TRUE)
+  
+  #create a list of all ens
+  nc_var_list <- map2(cf_var_names,
```

Comment:
Please use explicit namespacing.

```suggestion
  nc_var_list <- purrr::map2(cf_var_names,
```